akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerTestSetupBase.scala

```diff
@@ -71,4 +71,13 @@ abstract class HttpServerTestSetupBase {
   def send(string: String): Unit = send(ByteString(string.stripMarginWithNewline("\r\n"), "UTF8"))
 
   def closeNetworkInput(): Unit = netIn.sendComplete()
+
+  def shutdownBlueprint(): Unit = {
+    netIn.sendComplete()
+    requests.expectComplete()
+
+    responses.sendComplete()
+    netOut.expectBytes(ByteString("HTT")) // ???
```

Comment:
I have opened a separate ticket #20377 to look at this so we can get the resource leak fix in. Pretty sure the `HTT` behavior of the server in this case wasn't really introduced here, just noticed so it should be safe to merge.